SingularityService/src/main/java/com/hubspot/singularity/data/history/DeployTaskHistoryHelper.java

```diff
@@ -0,0 +1,112 @@
+import com.hubspot.singularity.SingularityTaskId;
+import com.hubspot.singularity.SingularityTaskIdHistory;
+import com.hubspot.singularity.config.SingularityConfiguration;
+import com.hubspot.singularity.data.TaskManager;
+
+@Singleton
+public class DeployTaskHistoryHelper extends BlendedHistoryHelper<SingularityTaskIdHistory, SingularityDeployKey> {
+
+  private final TaskManager taskManager;
+  private final HistoryManager historyManager;
+  private final SingularityConfiguration singularityConfiguration;
+
+  @Inject
+  public DeployTaskHistoryHelper(TaskManager taskManager, HistoryManager historyManager, SingularityConfiguration singularityConfiguration) {
+    this.taskManager = taskManager;
+    this.historyManager = historyManager;
+    this.singularityConfiguration = singularityConfiguration;
+  }
+
+  public List<SingularityTaskIdHistory> getHistoriesFor(Collection<SingularityTaskId> taskIds) {
+    Map<SingularityTaskId, SingularityTask> tasks = taskManager.getTasks(taskIds);
+    Map<SingularityTaskId, List<SingularityTaskHistoryUpdate>> map = taskManager.getTaskHistoryUpdates(taskIds);
+
+    List<SingularityTaskIdHistory> histories = Lists.newArrayListWithCapacity(taskIds.size());
+
+    for (SingularityTaskId taskId : taskIds) {
+      List<SingularityTaskHistoryUpdate> historyUpdates = map.get(taskId);
+      SingularityTask task = tasks.get(taskId);
+
+      histories.add(SingularityTaskIdHistory.fromTaskIdAndTaskAndUpdates(taskId, task, historyUpdates));
+    }
+
+    Collections.sort(histories);
+
+    return histories;
+  }
+
+  @Override
+  protected List<SingularityTaskIdHistory> getFromZk(final SingularityDeployKey deployKey) {
+    List<SingularityTaskId> requestTaskIds = taskManager.getTaskIdsForRequest(deployKey.getRequestId());
+
+    final Iterable<SingularityTaskId> deployTaskIds = Iterables.filter(requestTaskIds, new Predicate<SingularityTaskId>() {
+      @Override
+      public boolean apply(SingularityTaskId input) {
+        return input.getDeployId().equals(deployKey.getDeployId());
+      }
+    });
+
+    return getHistoriesFor(ImmutableList.copyOf(deployTaskIds));
+  }
+
+  @Override
+  protected List<SingularityTaskIdHistory> getFromHistory(final SingularityDeployKey deployKey, int historyStart, int numFromHistory) {
+    List<SingularityTaskIdHistory> requestHistories = historyManager.getTaskHistoryForRequest(deployKey.getRequestId(), historyStart, numFromHistory);
+
+    final Iterable<SingularityTaskIdHistory> deployHistories = Iterables.filter(requestHistories, new Predicate<SingularityTaskIdHistory>() {
+      @Override
+      public boolean apply(SingularityTaskIdHistory input) {
+        return input.getTaskId().getDeployId().equals(deployKey.getDeployId());
+      }
+    });
+
+    return ImmutableList.copyOf(deployHistories);
+  }
+
+  public List<SingularityTaskIdHistory> getActiveDeployTasks(SingularityDeployKey key) {
+    List<SingularityTaskIdHistory> histories = this.getFromZk(key);
```

Comment:
There won't be active tasks in mysql, we don't persist them if they are still active, still in lb cleanup, or part of a pending deploy. The task history by requestId does this the same way